contracts/Wrapper.sol

```diff
@@ -0,0 +1,84 @@
+// contracts/Wrapper.sol
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "./token/PartialCommonOwnership721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+
+struct WrappedToken {
+  address contractAddress;
+  uint256 id;
+}
+
+contract Wrapper is PartialCommonOwnership721 {
+  /// @notice Monotonic counter for token ids
+  uint256 private tokenIdCounter = 0;
```

Comment:
What about using something like:

```
  function onERC721Received(address operator, address from, uint256 tokenId, bytes memory _data) override public returns (bytes4) {
    // This assumes msg.sender is the NFT contract
    uint256 localTokenId = uint256(uint32(bytes4(keccak256(abi.encode(msg.sender, tokenId)))));
    _mint(_steward, localTokenId);
    return bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"));
  }
```
(1) the localTokenId is deterministic so might make it accessible for public/future use.
(2) minting is triggered automagically when the wrapper receives the token.

Reply:
I'm hoping for this project to become more collaborative and appreciate the suggestion!   Want to implement the changes in a child-branch and open a PR back into this branch?  If you do, please include acceptance tests within `tests/Wrapper/index.ts`.